lms/static/js/header/header.js

```diff
@@ -0,0 +1,80 @@
+/*
+    Ensuring collapsible and accessible components on multiple
+    screen sizes for the responsive lms header.
+*/
+
+$(document).ready(function() {
+    'use strict';
+    var $hamburgerMenu;
+    // Toggling visibility for the user dropdown
+    $('.toggle-user-dropdown').click(function() {
+        var $dropdownMenu = $('.global-header .nav-item .dropdown-user-menu');
+        if ($dropdownMenu.is(':visible')) {
+            $dropdownMenu.hide();
+        } else {
+            $dropdownMenu.show();
+            $dropdownMenu.find('.dropdown-item')[0].focus();
+        }
+    });
+
+    // Toggling menu visibility with the hamburger menu
+    $hamburgerMenu = $('.hamburger-menu');
+    $hamburgerMenu.click(function() {
+        $('.nav-links').toggleClass('nav-links-expanded');
+        $('.hamburger-menu').toggleClass('open');
+    });
+
+    // Hide hamburger menu if no nav items (sign in and register pages)
+    if ($('.nav-item').size() === 0) {
+        $hamburgerMenu.css('display', 'none');
+    }
+});
+
+// Ensure click away hides the dropdown
+$(window).click(function(e) {
+    'use strict';
+    if (!$(e.target).is('.dropdown-item, .toggle-user-dropdown')) {
+        $('.global-header .nav-item .dropdown-user-menu').hide();
+    }
+});
+
+// Accessibility keyboard controls for hamburger menu
+$(document).on('keydown', function(e) {
+    'use strict';
+    if (e.keyCode === 13 && $(e.target).hasClass('hamburger-menu')) {
+        $(e.target).click();
+    }
+});
+
+// Accessibility keyboard controls for user dropdown
+$(document).on('keydown', function(e) {
+    'use strict';
+    var isNext;
+    var isToggle = $(e.target).hasClass('toggle-user-dropdown');
+    var isDropdownOption = $(e.target).parent().hasClass('dropdown-item');
+
+    // Open dropdown menu on enter or space click
+    if ((e.keyCode === 13 || e.keyCode === 32) && isToggle) {
```

Comment:
You can import key code constants from the UI Toolkit. See an example here:

https://github.com/edx/edx-platform/blob/9c6d94ada0f2b401aeb5c41b3dbf31a912a80840/openedx/features/course_experience/static/course_experience/js/CourseOutline.js#L18